app/assets/javascripts/shared_functions.js

```diff
@@ -0,0 +1,16 @@
+function printErrors(error){
```

Comment:
This is some pretty complicated code for just printing out strings. Rather than this, I would suggest a convention of using `model.errors.full_messages` in controllers and just having this iterate through those.

Reply:
updated

Reply:
I've added the need migration, feel free to review the PR again.